src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,167 @@
+    }
+
+    function maxBar() public view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 stableDebtTotal = stableDebt.totalSupply();
+        uint256 variableDebtTotal = variableDebt.totalSupply();
+        return _calculateTargetSupply(targetInterestRate, stableDebtTotal, variableDebtTotal);
+    }
+
+    function _calculateTargetSupply(uint256 targetInterestRate, uint256 stableDebtTotal, uint256 variableDebtTotal) internal view returns (uint256) {
+        uint256 base = interestStrategy.baseVariableBorrowRate();
+        require(targetInterestRate > base, "DssDirectDepositAaveDai/target-interest-base");
+        require(targetInterestRate <= interestStrategy.getMaxVariableBorrowRate(), "DssDirectDepositAaveDai/above-max-interest");
+
+        // Do inverse calculation of interestStrategy
+        uint256 variableRateSlope1 = interestStrategy.variableRateSlope1();
+        uint256 targetUtil;
+        if (targetInterestRate > _add(base, variableRateSlope1)) {
+            // Excess interest rate
+            uint256 r = targetInterestRate - base - variableRateSlope1;
+            targetUtil = _add(_rdiv(_rmul(interestStrategy.EXCESS_UTILIZATION_RATE(), r), interestStrategy.variableRateSlope2()), interestStrategy.OPTIMAL_UTILIZATION_RATE());
+        } else {
+            // Optimal interest rate
+            targetUtil = _rdiv(_rmul(_sub(targetInterestRate, base), interestStrategy.OPTIMAL_UTILIZATION_RATE()), variableRateSlope1);
+        }
+        return _rdiv(_add(stableDebtTotal, variableDebtTotal), targetUtil);
+    }
+
+    function getTargetAssets(uint256 currentAssets) external override view returns (uint256) {
+        uint256 targetInterestRate = bar;
+        if (targetInterestRate == 0) return 0;     // De-activated
+
+        uint256 totalPoolSize = _add(
+                TokenLike(dai).balanceOf(address(adai)),
+                _add(
+                    stableDebt.totalSupply(),
+                    variableDebt.totalSupply()
+                )
+            );
+
+        uint256 stableDebtTotal = stableDebt.totalSupply();
+        uint256 variableDebtTotal = variableDebt.totalSupply();
+
+        uint256 targetTotalPoolSize = _calculateTargetSupply(targetInterestRate, stableDebtTotal, variableDebtTotal);
+        if (targetTotalPoolSize >= totalPoolSize) {
+            // Increase debt (or same)
+            return _add(currentAssets, targetTotalPoolSize - totalPoolSize);
+        } else {
+            // Decrease debt
+            uint256 decrease = totalPoolSize - targetTotalPoolSize;
+            if (currentAssets >= decrease) {
+                return currentAssets - decrease;
+            } else {
+                return 0;
+            }
+        }
+    }
+
+    function disable() external override auth {
+        bar = 0;
```

Comment:
This should emit an event.

Reply:
The hub emits an event, that's why I wasn't sure about this one. No harm in adding it here as well I guess.

Reply:
Makes more sense to have it in the plan imo, but I'm not too picky.